rfcs/min-relations.md

```diff
@@ -422,11 +422,27 @@ We can even combine positive and negative filters by including multiple copies o
 This advanced technique is helpful when you really need to be specific, as shown in these examples:
 
 ```rust
-fn herbivores(){
-
+fn herbivores(
+    commands: mut Commands,
+    plants_query: Query<Entity, With<Plant>>,
+    animals_query: Query<Enity, With<Animal>>,
+    species_query: Query<Entity, (With<Relation<Consumes>, Without<Relation<Consumes>>)>
+){
+    let plants = plants_query.iter().collect();
+    let animals = animals_query.iter().collect();
+
+    species_query
+        // The second type parameter controls which relation filter of that type is being modified
+        .filter_relation::<Consumes, InFilter<{1}>>(RelationFilter::any_of().targets(plants))
```

Comment:
I'm not sure about this one. Adding extra generic param to that function call will always make it required to list, even if this is unambiguous, e.g. like thtat
```rust
        .filter_relation::<Owes, _>>(target)
```
which just doesn't look good. Preferably we would structure it such that this generic can be auto-inferred somewhere, which should make the compiler solve the unambiguous case (given relation type is used only once in the query) without any extra help.

I know it's possible, because that's how `frunk` crate infers the `Indices` to operate on. You don't have to do anything if there is only one valid and provable set of indices for given operation. But it might require some extra gymnastics around generics for the relation filter itself or something. Also it would be nice if the index could be specified at runtime, but this is secondary.

Reply:
Alternative syntax that avoid `_`:
```rust
        .filter_relation::(RelationFilter::<Owes>::target(target))
```

Reply:
This second type parameter is definitely my largest complaint with the current design as well.